app/src/main/java/org/kinecosystem/kinit/view/MainActivity.java

```diff
@@ -47,22 +58,56 @@ protected void onCreate(Bundle savedInstanceState) {
         bottomTabNavigation = findViewById(R.id.navigation);
         int selectedTabIndex = (savedInstanceState != null) ? savedInstanceState.getInt(SELECTED_TAB_INDEX_KEY, 0) : 0;
         bottomTabNavigation.setSelectedTabIndex(selectedTabIndex);
-        enablePageSelection();
         tabsAdapter = new TabsAdapter();
-        tabsAdapter.setPageSelectionListener(this);
         viewPager = findViewById(R.id.view_pager);
         viewPager.setAdapter(tabsAdapter);
-        if (!tabsAdapter.shouldShowAnimation()) {
-            viewPager.setCurrentItem(TabsAdapter.EARN_TAB_INDEX, false);
+        enablePageSelection();
+        if (shouldShowPreEarnAnimation()) {
+            showPreEarnAnimation();
+        } else {
+            View readyToEarnText = findViewById(R.id.ready_to_earn_image);
+            ((ViewGroup) readyToEarnText.getParent()).removeView(readyToEarnText);
         }
         handleIntentExtras(getIntent().getExtras());
     }
 
+    private boolean shouldShowPreEarnAnimation() {
+        return taskRepository.isTaskAvailable() && !taskRepository.isTaskStarted().get();
+    }
+
+    private void showPreEarnAnimation() {
+        disablePageSelection();
+        View viewPager = findViewById(R.id.view_pager);
+        View preEarnImage = findViewById(R.id.ready_to_earn_image);
+        viewPager.setVisibility(View.INVISIBLE);
+        Animation slideLeftIn = AnimationUtils.loadAnimation(this, R.anim.slide_left_in);
+        Animation slideLeftOut = AnimationUtils.loadAnimation(this, R.anim.slide_left_out);
+
+        slideLeftIn.setDuration(PRE_ANIMATION_DURATION);
+        slideLeftOut.setDuration(PRE_ANIMATION_DURATION);
+
+        slideLeftIn.setInterpolator(new AccelerateDecelerateInterpolator());
+        slideLeftOut.setInterpolator(slideLeftIn.getInterpolator());
+
+        slideLeftIn.setAnimationListener(this);
+
+        viewPager.postDelayed(() -> {
+            findViewById(R.id.view_pager).setVisibility(View.VISIBLE);
```

Comment:
Can you just use the `viewPager` instance?

Reply:
can the user navigate to another page, before this runnable get executed? what will be the user experience in this way ?.
maybe you should check the page index?

Reply:
can't, nav gets locked on line 79